### ISSUE TITLE
Importing winauth.xml broken

### ISSUE THREAD

**Issue by c0rn3j** [winauth/winauth#416]
Using Add>Import only imports the first authenticator in the list.
Tried with 4 google auths and steam+google auth, same result.

**Comment by winauth** [comment]
See post on groups.

Where was the import from? Same version?

**Comment by c0rn3j** [comment]
I read the post. 

The import was from 3.1.5. Now it might have been created in the .NET3.5 version or the 4.5, as I was trying both in WINE. I am now only using the 4.5 one. 
I then tried importing the WINE xml in windows, and it didn't load all the authenticators. Interestingly enough, I then tried using 1 google acc + steam only in the windows version, grabbed the .xml file and imported that back again in the linux version, only to find out that again only 1 was imported.


Also I would really appreciate if you documented that you save database backups in the registry 
HKEY_CURRENT_USER\Software\WinAuth3\Backup with what is apparently your PGP key. It is not mentioned in the FAQ at all, and only very briefly mentioned in the lost password article.

By the way importing the steam account works under WINE and the steam auth then works perfectly, even trade confirmation.

**Comment by c0rn3j** [comment]
I just have one big issue, I've deleted the .xml file, deleted the registry keys but it still loads db on startup. I am not sure if this is some weird linux caching thing, some weird WINE caching thing or WinAuth is loading the file from the void.

**Comment by winauth** [comment]
Well, it's a bug in the import. Looks like it was from the 2.x days when there was only one authenticator per file. I have fixed it here and will try and get it updated soon. In the meantime, you can use the "Export" and then import that. It should get them all that way.

Yes, registry backup with my PGP key isn't documented enough. It isn't ideal and was meant as a last resort backup. However, it has saved data from some users. There is an option to put your own key into the config. Ideally, it would backup to the cloud, but my experience is people wouldn't do that either!

Check you don't have another winauth.xml is the same folder. It checks the current folder and then the AppData folder.

**Comment by c0rn3j** [comment]
> you can use the "Export" 
Yeah that's one of the things that crashes under WINE... I've needed it for importing the steam auth from windows, and I've already achieved that by importing it as the first item in the .xml

>Check you don't have another winauth.xml is the same folder. It checks the current folder and then the AppData folder.
There is nothing with the name of "winauth" in the whole WINE folder, except for the folder in application data :/
Is there a way to check where did it load the data from?

**Comment by winauth** [comment]
You would have to run something that watched file activity to see where it opened it from.

The config file is either an argument passed to winauth, loaded from current or AppData folder. There is also an upgrade from an old v2 version, which reads the registry key in HKLM\SOFTWARE\WinAuth\File1 and loads whatever is there.

If you turn encryption off, you can just open the xml file in a text editor and copy the <WinAuthAuthenticator> nodes.

**Comment by c0rn3j** [comment]
Using strace (described here http://unix.stackexchange.com/a/139031 ) I was able to track down my problem... 
It was my own idiocy because I did not set the current working directory to the path from where I launched the program, so it instead saved details to my home folder.

3368  open("/home/c0rn3j/.wineWinAuth32/dosdevices/z:/home/c0rn3j/winauth.xml", O_RDONLY|O_NONBLOCK) = 151


While you can set the PGP key to your own for the registry backups, I'd welcome if I could disable this behavior completely from within the GUI.

**Comment by winauth** [comment]
Glad you sorted it.

>  I'd welcome if I could disable this behavior completely from within the GUI.

If you run it in "portable mode" where the config file is taken from the same directory as the exe, then it will only write into that file, not to the registry.

**Comment by WinAuthFan** [comment]
@C0rn3j - you are running it under Linux? How? I can't get it to start. Get this error: 

```
$ mono WinAuth.exe 

Unhandled Exception:
System.NotImplementedException: The requested feature is not implemented.
  at System.Security.Principal.WindowsIdentity.get_User () [0x00000] in <filename unknown>:0 
  at WinAuth.SingleGlobalInstance.InitMutex () [0x00000] in <filename unknown>:0 
  at WinAuth.SingleGlobalInstance..ctor (Int32 timeOut) [0x00000] in <filename unknown>:0 
  at WinAuth.WinAuthMain.Main () [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.NotImplementedException: The requested feature is not implemented.
  at System.Security.Principal.WindowsIdentity.get_User () [0x00000] in <filename unknown>:0 
  at WinAuth.SingleGlobalInstance.InitMutex () [0x00000] in <filename unknown>:0 
  at WinAuth.SingleGlobalInstance..ctor (Int32 timeOut) [0x00000] in <filename unknown>:0 
  at WinAuth.WinAuthMain.Main () [0x00000] in <filename unknown>:0 
```

**Comment by C0rn3j** [comment]
@WinAuthFan I use WINE, not Mono.

https://appdb.winehq.org/objectManager.php?sClass=application&iId=17681
